### PR TITLE
Handle Speculos not replying outdated events while it performs OCR

### PR DESCRIPTION
**Pull request by fbeutin-ledger**

### CODE REVIEW

File: src/ragger/backend/speculos.py [hunk 1]
```diff
@@ -161,6 +167,7 @@ def both_click(self) -> None:
         self._client.press_and_release("both")
 
     def finger_touch(self, x: int = 0, y: int = 0, delay: float = 0.5) -> None:
+        self.logger.info(f"Sending finger_touch x={x}, y={y}, delay={delay}")
```
Comment (xchapron-ledger) [inline src/ragger/backend/speculos.py:170]: Maybe we should add an equivalent logs for button press?

File: src/ragger/backend/speculos.py [hunk 2]
```diff
@@ -82,17 +82,23 @@ def url(self) -> str:
         return f"http://{self._host}:{self._port}"
 
     def _retrieve_client_screen_content(self) -> dict:
-        raw_content = self._client.get_current_screen_content()
+        raw_events = self._client.get_current_screen_content().get("events", [])
+        while raw_events == []:
```
Comment (xchapron-ledger) [inline src/ragger/backend/speculos.py:86]: What if the screen doesn't have text content? Like only glyphs ?

File: src/ragger/backend/speculos.py [hunk 3]
```diff
@@ -82,17 +82,23 @@ def url(self) -> str:
...
+            print(f"raw_events = '{raw_events}'")
```
Comment (lpascal-ledger) [inline src/ragger/backend/speculos.py:87]: To be cleaned?

File: src/ragger/backend/speculos.py [hunk 4]
```diff
@@ -82,17 +82,23 @@ def url(self) -> str:
...
         # They probably comes from long press progress bar on Stax
         # and if not removed they falsely make wait_for_screen_change
         # consider screen as changed when screen text didn't and that's
         # what we want here.
         events = []
-        for event in raw_content.get("events", []):
+        for event in raw_events:
             if event.get("text", "").strip():
                 events.append(event)
+        self.logger.info(f"Received '{events}'")
```
Comment (lpascal-ledger) [inline src/ragger/backend/speculos.py:101]: Same as on Speculos: `.debug` rather? And f-strings.

File: src/ragger/backend/speculos.py (hunk 1 above)
Comment (lpascal-ledger) [inline src/ragger/backend/speculos.py:170]: Same as before: `.debug` rather? And f-strings.

**Comment by fbeutin-ledger** [comment]
> Is this PR still relevant?

Nope

**Pull request closed without merging**